### PR TITLE
FEAT: post requests validation

### PR DESCRIPTION
**Pull request by baimamboukar**
### Changes made for #13 
✅ Added validation logic for Experience, Education, and Skill POST requests.
✅  Implemented custom validation for grade, experience, and date fields.
✅ Returned appropriate error messages and HTTP status codes for invalid requests.
✅  Added a test function in `test_pytest.py` to verify the validation behavior.

@dtemir Thanks for reviewing 🫡

### CODE REVIEW

File: test_pytest.py [hunk 1]
```diff
@@ -4,71 +4,116 @@
...
+        "proficiency": "15%",
+        "logo": "example-logo.png"
+    })
+    print(response.json)
+    assert response.status_code == 201
+
+if __name__ == '__main__':
+    test_add_experience()
+    test_add_education()
+    test_add_skill()
```
Comment (dtemir) [inline test_pytest.py:118]: This part is not necessary because it's already automated with Pytest

File: utils.py
Comment (dtemir) [inline utils.py:1]: this is great! good job with `datetime` library

**Review by dtemir — CHANGES_REQUESTED** [review]

File: test_pytest.py (hunk 1 above)
Comment (baimamboukar) [reply test_pytest.py:118]: Okay alright! I will add a commit that removes this part 🚀

File: utils.py
Comment (baimamboukar) [reply utils.py:1]: Thanks 🫡

**Comment by baimamboukar** [comment]
@dtemir I removed the unnecessary parts ✅

**Comment by dtemir** [comment]
great job!

**Comment by winstxnhdw** [comment]
Hey @baimamboukar, I took a look at your changes to the tests, and from what I can tell, the test will not work at all..

**Comment by dtemir** [comment]
Yep, let's get them fixed

**Pull request merged**